addons/purchase/tests/test_purchase_invoice.py

```diff
@@ -229,3 +230,48 @@ def test_vendor_bill_ordered_return(self):
         for line in purchase_order.order_line:
             self.assertEqual(line.qty_to_invoice, 0.0)
             self.assertEqual(line.qty_invoiced, 10)
+
+    def test_vendor_severals_bills_and_multicurrency(self):
+        """
+        This test ensures that, when adding several PO to a bill, if they are expressed with different
+        currency, the amount of each AML is converted to the bill's currency
+        """
+        PurchaseOrderLine = self.env['purchase.order.line']
+        PurchaseBillUnion = self.env['purchase.bill.union']
+        ResCurrencyRate = self.env['res.currency.rate']
+        usd = self.env.ref('base.USD')
+        eur = self.env.ref('base.EUR')
+        purchase_orders = []
+
+        ResCurrencyRate.create({'currency_id': usd.id, 'rate': 1})
+        ResCurrencyRate.create({'currency_id': eur.id, 'rate': 2})
+
+        for currency in [usd, eur]:
+            po = self.env['purchase.order'].with_context(tracking_disable=True).create({
+                'partner_id': self.partner_a.id,
+                'currency_id': currency.id,
+            })
+            pol_prod_order = PurchaseOrderLine.create({
+                'name': self.product_order.name,
+                'product_id': self.product_order.id,
+                'product_qty': 1,
+                'product_uom': self.product_order.uom_id.id,
+                'price_unit': 1000,
+                'order_id': po.id,
+                'taxes_id': False,
+            })
+            po.button_confirm()
+            pol_prod_order.write({'qty_received': 1})
+            purchase_orders.append(po)
+
+        move_form = Form(self.env['account.move'].with_context(default_move_type='in_invoice'))
+        move_form.purchase_vendor_bill_id = PurchaseBillUnion.browse(-purchase_orders[0].id)
+        move_form.purchase_vendor_bill_id = PurchaseBillUnion.browse(-purchase_orders[1].id)
```

Comment:
Why a negative `id` for the browse ?

Reply:
This is how a part of the view is defined
https://github.com/odoo/odoo/blob/81a8be7d115ffcfac0126c252811381a482bd6a9/addons/purchase/report/purchase_bill.py#L34-L41